internal/exec/util/file.go

```diff
@@ -142,52 +128,20 @@ func RenderFile(l *log.Logger, c *resource.HttpClient, f types.File) *File {
 
 	return &File{
 		Path:        f.Path,
-		ReadCloser:  reader,
-		Hash:        fileHash,
+		Hash:        hasher,
 		Mode:        os.FileMode(f.Mode),
 		Uid:         *f.User.ID,
 		Gid:         *f.Group.ID,
 		expectedSum: expectedSum,
+		Url:         *u,
+		Client:      c,
+		Compression: f.Contents.Compression,
+		AWSRegion:   f.Contents.Aws.Region,
 	}
 }
 
-// gzipReader is a wrapper for gzip's reader that closes the stream it wraps as well
-// as itself when Close() is called.
-type gzipReader struct {
-	*gzip.Reader //actually a ReadCloser
-	source       io.Closer
-}
-
-func newGzipReader(reader io.ReadCloser) (io.ReadCloser, error) {
-	gzReader, err := gzip.NewReader(reader)
-	if err != nil {
-		return nil, err
-	}
-	return gzipReader{
-		Reader: gzReader,
-		source: reader,
-	}, nil
-}
-
-func (gz gzipReader) Close() error {
-	if err := gz.Reader.Close(); err != nil {
-		return err
-	}
-	if err := gz.source.Close(); err != nil {
-		return err
-	}
-	return nil
-}
-
-func decompressFileStream(l *log.Logger, f types.File, contents io.ReadCloser) (io.ReadCloser, error) {
-	switch f.Contents.Compression {
-	case "":
-		return contents, nil
-	case "gzip":
-		return newGzipReader(contents)
-	default:
-		return nil, types.ErrCompressionInvalid
-	}
+func intToPtr(i int) *int {
```

Comment:
Didn't we added an `IntToPtr` to internal/util/util.go already? Seems you could use that.